python/cuml/dask/common/base.py

```diff
@@ -18,27 +18,78 @@
 import numpy as np
 from toolz import first
 
-import cudf.comm.serialize  # noqa: F401
+from cuml.dask.common.utils import get_client
+from cuml.dask.common.input_utils import get_datatype
+import pickle
 
 from cuml import Base
 from cuml.common.array import CumlArray
 
 from dask_cudf.core import DataFrame as dcDataFrame
 
-from dask.distributed import default_client
 from functools import wraps
 
 
 class BaseEstimator(object):
 
-    def __init__(self, client=None, verbose=False, **kwargs):
+    def __init__(self, client=None, verbose=False, model=None, **kwargs):
         """
         Constructor for distributed estimators
         """
-        self.client = default_client() if client is None else client
+        self.client = get_client(client)
         self.verbose = verbose
         self.kwargs = kwargs
 
+        self.set_model(model)
+
+    @classmethod
+    def load(cls, file, client=None, verbose=False, pickle_args={}, **kwargs):
+        """
+        Convenience function to load a saved cuml model into a distributed
+        model.
+        """
+        model = pickle.load(file, **pickle_args)
+
+        if isinstance(model, Base):
+            # If serialized model is single GPU, create new
+            # dist model
+            model = cls.__init__(client=client,
```

Comment:
I'm torn about this... I worry that it's a recipe to create "weird" distributed models that aren't properly populated, though they have a nice `local_model`.
E.g. if I have a saved SG random forest with `n_estimators == 50` and I load that into an MG random forest, what is that object's `self.n_estimators`? It seems like the caller would be responsible for pushing those all through kwargs, or you end up with weird missing fields (which may be confusing).

Dask-ML has a way of turning SG models into MG for inference only based on their parallel meta estimators: https://ml.dask.org/meta-estimators.html

I think one nice property there is that you clearly get an object that is a MetaEstimator wrapped around a local estimator - it's clearly a bit different from a true MG model. Maybe there is some way we can convey that here - with a separate class or some other clear indicator that it is inference only?

Actually, if you're purely using this local model for embarassingly-parallel prediction, then we are basically just using the mixin as a standalone and the actual distributed class may be irrelevant?

Reply:
In the pytests, I managed to demonstrate the sg->mg and mg->sg features without even needing this function. What do you say we ditch the static load function? Besides running the risk of becoming inflexible, I think we both agree this could quickly become a maintenance burden. 

I think I like the idea of the Dask wrappers accepting a `model` argument more than this, because it still allows the user to unpickle an sg model and use it for distributed inference without cuml having to worry about how the model was pickled.